### PR TITLE
Feature/hdf5

### PR DESCRIPTION
**Pull request by hroest**
add option to build HDF5 libraries. This will become important in the future for OpenMS as well, so we should at least have the option to link against HDF5. Currently there is mz5 and mzMLb (binary) that use HDF5 and other file formats are on the way that use it. I think we should start thinking about supporting them as they have some very nice features (including fast access, lower disk usage, option to build index trees on top of the data structure etc).

Currently, it requires the following library to be copied manually into `archives`

https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.5/src/hdf5-1.10.5.tar.gz

@jpfeuffer can you upload this to the FTP location where all the other source package are?

PR testing:
- Tested on Linux
- Tested on Windows

feel free to test this, also test on OSX. Currently on Linux we build shared and static libraries (.so and .a), on Windows it just builds the .lib files. I am not sure whether that is okay with Windows guys but probably enough to get it working, @cbielow any comments?

### CODE REVIEW

File: libraries.cmake/hdf5.cmake [hunk 1]
```diff
@@ -0,0 +1,90 @@
+##################################################
+###       HDF5                                 ###
+##################################################
+MACRO( OPENMS_CONTRIB_BUILD_HDF5 )
+
+  # Minimal cmake requirement for building HDF5 1.10.5
+  cmake_minimum_required(VERSION 3.10.0 FATAL_ERROR)
```
Comment (hroest) [inline libraries.cmake/hdf5.cmake]: this may be a slight hickup, @jpfeuffer will this cause any problems?

Comment (jpfeuffer) [reply libraries.cmake/hdf5.cmake]: Hmm not necessarily. I am okay with upgrading to newer CMake's (I don't think all our build machine fulfill this requirement right now).

**Comment by jpfeuffer** [comment]
Done updating. Re-running to see if more changes are required.

**Comment by hroest** [comment]
@jpfeuffer merge?

**Comment by cbielow** [comment]
this PR destroys Windows debug builds.
` libhdf5_cpp.lib(H5FaccProp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in MockImplementation.obj` ... many of those....

The reason is the missing debug library. Linking the release lib against a debug build will cause the above error.

**Comment by cbielow** [comment]
We should REALLY have debug builds for Windows by default to catch this.

**Comment by jpfeuffer** [comment]
We have nightly. See https://github.com/OpenMS/OpenMS/issues/4091

**Pull request merged**